From 2ae0b82e882bfcc50d40b59c86ed4874a2b3d92a Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Wed, 8 Nov 2023 18:56:07 +0530
Subject: [PATCH] pcie: qcom: Update domain check

While switching PCIE2/3 to single
lane in alder, check for domain 2
and 3 as the pci domain index starts
from 0.

Change-Id: I6a07f4cdf822bd662a06d893950caee2ac1e64f7
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 drivers/pci/controller/dwc/pcie-qcom.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index d3af320f9698..4da3b94edf22 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -2076,7 +2076,7 @@ static void qcom_ipq_switch_lane(struct pci_dev *dev)
 	np = devp->of_node;
 
 	/* Switching PCIE Nodes 2/3 to single lane if force_to_single_lane property is defined in dts */
-	if ((of_property_read_bool(np, "force_to_single_lane")) && (pcie->domain == 3 || pcie->domain == 4)) {
+	if ((of_property_read_bool(np, "force_to_single_lane")) && (pcie->domain == 2 || pcie->domain == 3)) {
 
 		dev_info(devp,"Forcing PCIE to single lane\n");
 
-- 
2.34.1

